### DISCUSSION TITLE
Switching between wifi_ap and esp_now

### DISCUSSION THREAD

**Discussion by orbitcoms** [Mair/esp32-course#172] [Course Qeustions]
Hi,
I am wanting to switch between wifi as AP and overe to ESP_Now and later back again etc.
If I start with say wifi initialiazed and running as AP, do I simply do esp_wifi_disconnect() followed by esp_wifi_stop() then can init the espo now and establish connection and to go back I stop the esp-now and run the wifi_ap config again (ie: run wifi init following by wifi_connect_ap). 
Will this work, I can re-call the init and connection foir wifi and same with ESP-now over and over as long as I first stop the wifi or esp-now? Or do I also have to do something with the oibject that was configured to be able to run the configuration again? 

In addition to my post. Dio I need to turn the wifi off at all? Can I just use my WIFI_AP connecvtion for ESP_NOW as well?
My project only needs to send broadcast data now and then. The AP is serving a web page

**Comment by Mair** [comment]
Hi @orbitcoms,
Some room for experiments here. I haven't attempted this. The results would be interesting
